### PR TITLE
Stcom auto completion

### PR DESCRIPTION
**Pull request by aditya-matukumalli**

### CODE REVIEW

**Comment by cherewaty** [comment]
A BigTest! Nice! 🎉 

A handy tool you can use to see how comprehensive your tests are:
- change 
```describe.only('AutoComplete', () => {``` 
to 
```describe('AutoComplete', () => {```
 (we don't want that `.only()` in `master` anyway, since it'll prevent any other tests from running)
- run `yarn test --coverage`
- open `artifacts/coverage/lcov-report/lib/AutoComplete/AutoComplete.js.html` in a browser
- it'll show you the lines that could still use some coverage; there aren't many - a few more assertions could get it to 100%

**Comment by aditya-matukumalli** [comment]
@cherewaty Ah! didnt see this comment until now, my bad. 
image
Added in an image that shows the coverage following your instructions. Does that mean 92% lines are covered or need coverage?

File: lib/AutoComplete/AutoComplete.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+class AutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    this.textfield = React.createRef();
+    this.onOuterClick = this.onOuterClick.bind(this);
+  }
+
+  // To override onOuterClicks default behavior of reseting the input value
+  onOuterClick(e) {
+    this.props.onChange(e.inputValue);
```
Comment (taras) [inline lib/AutoComplete/AutoComplete.js]: `onChange` prop should always send unmodified event. If want to send value, then can do something similar to what Downshift is doing which have a special prop called onInputValueChange.

File: lib/AutoComplete/AutoComplete.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+              </div>
+              )
+            }
+        </Downshift>
+      </TetherComponent>
+    );
+  }
+}
+
+AutoComplete.propTypes = propTypes;
```
[the remark this answers is not in the document]
Comment (aditya-matukumalli) [reply lib/AutoComplete/AutoComplete.js]: @taras Redux-form tests u mean, similar to https://github.com/folio-org/stripes-components/blob/master/lib/TextField/tests/TextField-ReduxForm-test.js?

**Comment by aditya-matukumalli** [comment]
Closing this for now. As the autoComplete here serves mainly a smaller subset of purposes pertaining to a specific requirement, and for it to be a part of the shared component library, needs more comprehensive developer controls to be added  to expand its usefulness. This can be revisited in the near future

**Pull request closed without merging**